### PR TITLE
Neural Solution Resource Management

### PR DESCRIPTION
**Pull request by Kaihui-intel**
## Type of Change

feature 
API changed or not: Neural Solution's server-side API

## Description
Resource management for Neural Solution maintainer.
usage: neural_solution cluster [action] (none/args)

- --query
- --join "host1 2 20; host 2 20"(split by ";") or hostfile
- --rm (node-id)

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

`prettytable` already in https://github.com/intel/neural-compressor/blob/6c30464a2a48d62be51cbe9f90f7cbb8d81f6edb/requirements.txt#L13

### CODE REVIEW

File: neural_solution/docs/source/README.md [hunk 1]
```diff
@@ -10,6 +10,11 @@
   - [Query task status](#query-task-status)
   - [Stop service](#stop-service)
   - [Inspect logs](#inspect-logs)
+  - [Start resource management](#start-resource-management)
```
Comment (yiliu30) [inline neural_solution/docs/source/README.md:13]:
```suggestion
  - [Start resource management](#start-resource-management)
  + [Manage resource](#manage -resource)  
```

File: neural_solution/docs/source/README.md [hunk 2]
```diff
@@ -126,3 +131,32 @@ There are several logs under workspace:
...
+```
+### Add node
+This command is used to add nodes to the cluster. You can either specify a host file or provide a list of nodes separated by ";". For example:
+```shell
+neural_solution cluster --join "host1 2 20; host2 4 20"
+```
+### Remove node
+This command is used to remove nodes from the cluster based on the IDs obtained from the query. The IDs can be passed as a parameter to the command. For example:
+```shell
+neural_solution cluster --rm <query_id>
```
Comment (yiliu30) [inline neural_solution/docs/source/README.md:160]:
```suggestion
neural_solution cluster --rm <query_id>
neural_solution cluster --remove <query_id>
```

**Comment by yiliu30** [comment]
Pls add the usage to one of the E2E examples.

File: neural_solution/docs/source/README.md [hunk 3]
```diff
@@ -126,3 +131,32 @@ There are several logs under workspace:
...
+
+### Query cluster
+This command is used to query the current status of the cluster. No additional parameters are required, simply enter the following command:
+```shell
+neural_solution cluster --query
+```
+### Add node
+This command is used to add nodes to the cluster. You can either specify a host file or provide a list of nodes separated by ";". For example:
+```shell
+neural_solution cluster --join "host1 2 20; host2 4 20"
```
Comment (yiliu30) [inline neural_solution/docs/source/README.md:155]: Could you provide more details regarding the node format?  
e.g., `hostname number_of_sockets cores_per_socket`

File: neural_solution/examples/hf_models/README.md [hunk 4]
```diff
@@ -118,6 +122,19 @@ optional arguments:
...
 [user@server tf_example1]$ curl -X GET  http://localhost:8000/download/{task_id} --output quantized_model.zip
 # download quantized_model.zip
+```
+### Manage resource
+```shell
+# query cluster information
+neural_solution cluster --query
+
+# add new node into cluster
+# parameter: "<node1> <number_of_sockets> <number_of_threads>;<node2> <number_of_sockets> <number_of_threads>"
```
Comment (yiliu30) [inline neural_solution/examples/hf_models/README.md:132]: A good explanation for the node parameter format. I suggest adding it to the document as well.

**Review by yiliu30 — APPROVED** [review]
Others are LGTM.

**Review by chensuyue — APPROVED** [review]

**Pull request merged**